[quant] Compute scale and zero point automatically in testing::_quantize

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46229 [quant] Statically quantized LSTM
* #46426 [quant] Allow replacement of the custom modules in the "prepare" step
* #46228 [quant] Custom module fix
* **#46232 [quant] Compute scale and zero point automatically in testing::_quantize**
* #46235 [quant] Quantized flip dispatch

Differential Revision: [D24271061](https://our.internmc.facebook.com/intern/diff/D24271061)

Commit:
[quant] Compute scale and zero point automatically in testing::_quantize

[ghstack-poisoned]

Commit:
Update on "[quant] Compute scale and zero point automatically in testing::_quantize"


Differential Revision: [D24271061](https://our.internmc.facebook.com/intern/diff/D24271061)

[ghstack-poisoned]

Commit:
Update on "[quant] Compute scale and zero point automatically in testing::_quantize"


Differential Revision: [D24271061](https://our.internmc.facebook.com/intern/diff/D24271061)

[ghstack-poisoned]

Commit:
Update on "[quant] Compute scale and zero point automatically in testing::_quantize"


Differential Revision: [D24271061](https://our.internmc.facebook.com/intern/diff/D24271061)

[ghstack-poisoned]

supriyar commented on torch/testing/_internal/common_quantized.py:
```diff
@@ -21,16 +21,27 @@ def _conv_output_shape(input_size, kernel_size, padding, stride, dilation,
     qx = np.clip(qx, qmin, qmax)
     qx = qx.astype(dtype)
-    return qx
+    return qx, scale, zero_point
```
Are we actually using the returned scale and zero_point anywhere in this PR?

z-a-f replied:
Only in one case I needed it -- this is optional, but because the current function returns the `int_repr` of the x, it losses information about the scale and zero_point, especially if it is inferred internally in this function

supriyar commented on torch/testing/_internal/common_quantized.py:
```diff
@@ -21,16 +21,27 @@ def _conv_output_shape(input_size, kernel_size, padding, stride, dilation,
 # Quantization references
-def _quantize(x, scale, zero_point, qmin=None, qmax=None, dtype=np.uint8):
+def _quantize(x, scale=None, zero_point=None, qmin=None, qmax=None,
+              dtype=np.uint8):
```
Might be a good idea to update the docstring for this function to reflect the new behavior.

z-a-f replied:
Do you mean the comment here? This is just an internal function for testing -- do we have a docstring?

Commit:
Update on "[quant] Compute scale and zero point automatically in testing::_quantize"


Differential Revision: [D24271061](https://our.internmc.facebook.com/intern/diff/D24271061)

[ghstack-poisoned]
